src/controllers/expenses.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const expensesService = require('../services/expenses');
+const usersService = require('../services/users');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  let expenses = expensesService.getAll();
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === +userId);
+  }
+
+  if (categories) {
+    expenses = expenses.filter(({ category }) => (
+      categories.includes(category)
+    ));
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from).getTime();
+    const toDate = new Date(to).getTime();
+
+    expenses = expenses.filter(expense => {
+      const currentDate = new Date(expense.spentAt).getTime();
+
+      return fromDate <= currentDate && currentDate <= toDate;
+    });
```

Comment:
You already have some of this filtering logic inside your `expensesService.getAll`, please pass your query arguments there and move this logic there.

Reply:
Check other endpoints for this too.